src/ApiException.php

```diff
@@ -115,6 +115,40 @@ public static function createFromApiResponse(
         );
     }
 
+    /**
+     * For REST-based responses, the metadata does not need to be decoded.
+     *
+     * @param string $basicMessage
+     * @param int $rpcCode
+     * @param array|null $metadata
+     * @param \Exception $previous
+     * @return ApiException
+     */
+    public static function createFromRestApiResponse(
+        $basicMessage,
+        $rpcCode,
+        array $metadata = null,
+        \Exception $previous = null
+    ) {
+        if (empty($metadata)) {
+            $decodedMetadata = [];
+            return self::create(
+                $basicMessage,
+                $rpcCode,
+                $metadata,
+                $decodedMetadata,
+                $previous
+            );
+        }
+        return self::create(
+            $basicMessage,
+            $rpcCode,
+            $metadata,
+            $metadata,
+            $previous
+        );
```

Comment:
It might be better to simplify this to:
```suggestion
        return self::create(
            $basicMessage,
            $rpcCode,
            $metadata,
            is_null($metadata) ? [] : $metadata,
            $previous
        );
```

Reply:
Ah that is much cleaner - will do, thank you!

Reply:
Much cleaner, thank you! Done.